deep_qa/data/instances/tuple_inference_instance.py

```diff
@@ -0,0 +1,443 @@
+                   'num_slots': max_num_slots}
+        if max_word_length > 0:
+            lengths['word_character_length'] = max_word_length
+
+    @staticmethod
+    def get_lengths_from_indexed_tuples(indexed_tuples: List):
+        '''
+        Helper method for self.get_lengths().  This gets the max lengths (max number of slots and max
+        number of words per slot) for a list of indexed tuples.
+
+        Parameters
+        ----------
+        indexed_tuples: List[List[List[int]]]
+            The indexed tuples that we want to get max lengths from.
+
+        Returns
+        -------
+        max_num_slots: int
+            The max number of slots found.
+
+        max_slot_words: int
+            The max number of words found in any of the slots.
+
+        max_word_length: int
+            The length of the longest word, with a return value of -1 if the model isn't using character padding.
+        '''
+        max_num_slots = 0
+        max_slot_words = 0
+        max_word_length = 0
+        for indexed_tuple in indexed_tuples:
+            num_slots = len(indexed_tuple)
+            max_num_slots = max(max_num_slots, num_slots)
+            for slot in indexed_tuple:
+                word_sequence_lengths = IndexedInstance._get_word_sequence_lengths(slot)
+                max_slot_words = max(max_slot_words, word_sequence_lengths['word_sequence_length'])
+                if 'word_character_length' in word_sequence_lengths:
+                    max_word_length = max(max_word_length, word_sequence_lengths['word_character_length'])
+
+        return max_num_slots, max_slot_words, max_word_length
+
+    @overrides
+    def pad(self, max_lengths: Dict[str, int]):
+        """
+        Pads (or truncates) each indexed tuple in answers_indexed and background_indexed to specified lengths,
+        using the superclass methods ``pad_sequence_to_length`` (for padding number of options, number of tuples,
+        and number of slots per tuple) and ''pad_word_sequence`` (for padding the words in each slot), with the
+        following exception:
+            When padding the number of answer options, if there are more answers provided, here we don't
+            remove options, we raise an exception because we don't want to risk removing the correct answer in
+            a multiple choice setting.
+        """
+        desired_num_options = max_lengths['num_options']
+        desired_num_question_tuples = max_lengths['num_question_tuples']
+        desired_num_background_tuples = max_lengths['num_background_tuples']
+        desired_num_slots = max_lengths['num_slots']
+        desired_question_slot_length = max_lengths['question_slot_length']
+        desired_background_slot_length = max_lengths['background_slot_length']
+        desired_word_character_length = None
+        if 'max_word_character_length' in max_lengths:
+            desired_word_character_length = max_lengths['max_word_character_length']
```

Comment:
`word_character_length` and `word_sequence_length` are keys defined in the super classes, and so for the API to work correctly, you need to use those keys when you call `pad_word_sequence`.    Yes, you could feasibly just change the name from your internal key when you pass it in, but it'd be easier to understand what's going on if you just used the same name.